test/hotspot/jtreg/compiler/c2/irTests/TestVectorFPReduction.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.*;
+
+/*
+ * @test
+ * @bug 8320725
+ * @summary Ensure strictly ordered AddReductionVF/VD nodes are generated on SVE machines
+ * while being disabled on Neon
+ * @library /test/lib /
+ * @run driver compiler.c2.irTests.TestVectorFPReduction
+ */
+
+public class TestVectorFPReduction {
+
+    final private static int SIZE = 1024;
+
+    private static double[] da = new double[SIZE];
+    private static double[] db = new double[SIZE];
+    private static float[] fa = new float[SIZE];
+    private static float[] fb = new float[SIZE];
+    private static float fresult;
+    private static double dresult;
+
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Test
+    @IR(applyIfCPUFeatureAnd = {"asimd", "true", "sve", "false"}, failOn = {IRNode.ADD_REDUCTION_VF})
+    @IR(applyIfCPUFeature = {"sve", "true"}, counts = {"requires_strict_order", ">=1", IRNode.ADD_REDUCTION_VF, ">=1"},
+        failOn = {"no_strict_order"}, phase = CompilePhase.PRINT_IDEAL)
+    private static void testAddReductionVF() {
+        float result = 1;
+        for (int i = 0; i < SIZE; i++) {
+            result += (fa[i] + fb[i]);
+        }
+        fresult += result;
+    }
+
+    @Test
+    @IR(applyIfCPUFeatureAnd = {"asimd", "true", "sve", "false"}, failOn = {IRNode.ADD_REDUCTION_VD})
+    @IR(applyIfCPUFeature = {"sve", "true"}, counts = {"requires_strict_order", ">=1", IRNode.ADD_REDUCTION_VD, ">=1"},
+        failOn = {"no_strict_order"}, phase = CompilePhase.PRINT_IDEAL)
```

Comment:
Also: I realize that you only check for `asimd / sve` features. Can you also apply it for avx features?

Reply:
I am not sure which specific avx/sse features/versions support these operations. Is it ok to add - 
`applyIfCPUFeatureOr = {"sve", "true", "avx", "true"}` ? 
or maybe anyone with x86 knowledge who would work on adding strict ordering conditions in their backend (after this patch is merged) can also modify these two testcases accordingly? Also in the vectorapi case, I have included only specific shapes of Vectors where the Reduction nodes would be generated on aarch64. It may or may not be the case for x86 and they might have to add more tests accordingly.

Reply:
I don't fully know what is ok. I usually check this experimentally, and  just put the lowest feature that makes my test pass.

Do you have access to any x86 machine you could test this on?

Reply:
Oh ok. Yes I do have an x86 machine to test on.